src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -178,6 +188,60 @@ public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String d
         }
     }
 
+    public DRSAccessURL getAccessUrlForObjectId(AuthenticatedUserRequest authUser, String objectId, String accessId) {
+        DRSObject object = lookupObjectByDrsId(authUser, objectId, false);
+
+        DrsId drsId = drsIdService.fromObjectId(objectId);
+        Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
+        SnapshotProject snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshot.getId());
```

Comment:
It looks like this partly duplicates code in `lookupObjectByDrsId()`, which this method is calling already. It would better if the same data isn't computed twice. One way to do that is to have `lookupObjectByDrsId()` return a static class that contains all the data that this method needs.

Reply:
Yeah, that's not a bad idea, especially since this can put some load on the DB